src/main/io/osd.h

```diff
@@ -177,8 +182,15 @@ STATIC_ASSERT(OSD_WARNING_COUNT <= 16, osdwarnings_overflow);
 #define ESC_TEMP_ALARM_OFF INT8_MIN
 #define ESC_CURRENT_ALARM_OFF -1
 
+typedef struct osdElementConfig_s {
+    int8_t x;
+    int8_t y;
+    uint8_t flags;
+} osdElementConfig_t;
```

Comment:
Do we actually need 24 bits to encode this? Ideally we'd want to reduce this to 16 bits.

Reply:
@mikeller : This change may be only temporary,, going back to uint16_t when CLI `set` is smart enough.

@pkruger : Accessor marcos will make internal representation opaque to most of code, only CLI will use actual representation

Reply:
@ledvinap: If it's temporary we should try and avoid ever releasing it - forcing the user to twice re-create their OSD config for a temporary change and change back is not desirable.

@pkruger: There are 4 or 5 bits available in the uint16_t. I think this should be enough?

Reply:
Adding CLI `set` for structured data will be nice, but nontrivial ...

Reply:
Also, with structured data, forward migration may be quite easy (if comma-separated format is supported)

Reply:
@mikeller I think you're right, even 3 bit is enough for OSD profiles, anything more than that is hard to keep track of.  Also like DanNixon said, the selection would be on a radio switch so anything more than 3 does make sense.